webscrambles/src/net/gnehzr/tnoodle/server/webscrambles/ScrambleRequest.java

```diff
@@ -603,6 +388,331 @@ private static void addScrambles(PdfWriter docWriter, Document doc, ScrambleRequ
+                int xPos = left + x*lineWidth + (x+1)*excessX/(linesX+1);
+                int yPos = (withScrambles ? solutionBorderTop : scrambleBorderTop) - (y+1)*availableSolutionHeight/(linesY+1);
+                cb.moveTo(xPos, yPos);
+                cb.lineTo(xPos+lineWidth, yPos);
+                moveCount++;
+            }
+        }
+
+        float UNDERLINE_THICKNESS = 0.2f;
+        cb.setLineWidth(UNDERLINE_THICKNESS);
+        cb.stroke();
+
+        if (withScrambles) {
+            cb.beginText();
+            int availableScrambleSpace = right-left - 2*padding;
+            int scrambleFontSize = 20;
+            String scrambleStr = "Scramble: " + scramble;
+            float scrambleWidth;
+            do {
+                scrambleFontSize--;
+                scrambleWidth = bf.getWidthPoint(scrambleStr, scrambleFontSize);
+            } while(scrambleWidth > availableScrambleSpace);
+
+            cb.setFontAndSize(bf, scrambleFontSize);
+            int scrambleY = 3 + solutionBorderTop+(scrambleBorderTop-solutionBorderTop-scrambleFontSize)/2;
+            cb.showTextAligned(PdfContentByte.ALIGN_LEFT, scrambleStr, left+padding, scrambleY, 0);
+            cb.endText();
+
+            int availableScrambleWidth = right-rulesRight;
+            int availableScrambleHeight = gradeBottom-scrambleBorderTop;
+            Dimension dim = scrambleRequest.scrambler.getPreferredSize(availableScrambleWidth-2, availableScrambleHeight-2);
+            PdfTemplate tp = cb.createTemplate(dim.width, dim.height);
+            Graphics2D g2 = new PdfGraphics2D(tp, dim.width, dim.height, new DefaultFontMapper());
+
+            try {
+                Svg svg = scrambleRequest.scrambler.drawScramble(scramble, scrambleRequest.colorScheme);
+                drawSvgToGraphics2D(svg, g2, dim);
+            } catch (InvalidScrambleException e) {
+                l.log(Level.INFO, "", e);
+            } finally {
+                g2.dispose();
+            }
+
+
+            cb.addImage(Image.getInstance(tp), dim.width, 0, 0, dim.height, rulesRight + (availableScrambleWidth-dim.width)/2, scrambleBorderTop + (availableScrambleHeight-dim.height)/2);
+        }
+
+        ColumnText ct = new ColumnText(cb);
+
+        int fontSize = 15;
+        int marginBottom = 10;
+        int offsetTop = 27;
+        boolean showScrambleCount = withScrambles && scrambleRequest.scrambles.length > 1;
+        if(showScrambleCount) {
+            offsetTop -= fontSize + 2;
+        }
+
+        cb.beginText();
+        cb.setFontAndSize(bf, fontSize);
+        cb.showTextAligned(PdfContentByte.ALIGN_CENTER, globalTitle, competitorInfoLeft+(right-competitorInfoLeft)/2, top-offsetTop, 0);
+        offsetTop += fontSize + 2;
+        cb.endText();
+
+        cb.beginText();
+        cb.setFontAndSize(bf, fontSize);
+        cb.showTextAligned(PdfContentByte.ALIGN_CENTER, scrambleRequest.title, competitorInfoLeft+(right-competitorInfoLeft)/2, top-offsetTop, 0);
```

Comment:
This is something like "3x3x3: Fewest Moves Round 1", right? I think that in the interest of a fully generic FMC scramble sheet, we wouldn't say anything like "Round 1" or "Round 2".